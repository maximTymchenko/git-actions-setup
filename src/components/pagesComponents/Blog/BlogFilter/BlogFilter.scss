@import '../../../../styles/vars';

.blog-filter {
    width: 100%;
    max-width: 600px;
    margin: 50px auto 0;
    border-top: 2px solid $section__border;

    &__wrapper {
        display: flex;
        justify-content: space-between;
        margin-top: 15px;
    }

    &__span {
        font-family: 'OktaNeue-Regular';
        font-size: 15px;
        color: $text__body;
        display: block;
        margin-top: 10px;
    }

    &__list-title {
        font-family: 'OktaNeue-Black';
        font-size: 28px;
        font-weight: 900;
        color: $text;
    }
}
