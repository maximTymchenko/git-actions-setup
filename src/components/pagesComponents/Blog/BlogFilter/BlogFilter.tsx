import React, { FC, useState } from 'react';
import './BlogFilter.scss';
import FilterList from '../../../common/FilterList';

const BlogFilter: FC = () => {
    const [servicesList, setServicesList] = useState([
        { id: 1, value: 'Web development', isChecked: false },
        { id: 2, value: 'Cloud services', isChecked: false },
        { id: 3, value: 'Research', isChecked: false },
        { id: 4, value: 'Analytics', isChecked: false },
        { id: 5, value: 'Quality control', isChecked: false },
        { id: 6, value: 'Mobile development', isChecked: false },
    ]);

    const handleCheckChieldServicesElement = (event: any): void => {
        const listCopy = servicesList.slice();
        listCopy.forEach((services) => {
            if (services.value === event.target.value) {
                services.isChecked = event.target.checked;
            }
        });
        setServicesList(listCopy);
    };

    const [industriesList, setIndustriesList] = useState([
        { id: 1, value: 'Retail & E-commerce', isChecked: false },
        { id: 2, value: 'Construction', isChecked: false },
        { id: 3, value: 'Finance', isChecked: false },
        { id: 4, value: 'Media', isChecked: false },
        { id: 5, value: 'Other', isChecked: false },
    ]);

    const handleCheckChieldIndustriesElement = (event: any): void => {
        const listCopy = industriesList.slice();
        listCopy.forEach((industries) => {
            if (industries.value === event.target.value) {
                industries.isChecked = event.target.checked;
            }
        });
        setIndustriesList(listCopy);
    };

    const [expertiseList, setExpertiseList] = useState([
        { id: 1, value: 'SaaS product', isChecked: false },
        { id: 2, value: 'Chatbots', isChecked: false },
        { id: 3, value: 'AR & VR', isChecked: false },
        { id: 4, value: 'Blockchain', isChecked: false },
        { id: 5, value: 'Digital enterprise ', isChecked: false },
        { id: 6, value: 'Drone', isChecked: false },
        { id: 7, value: 'R&D', isChecked: false },
    ]);

    const handleCheckChieldExpertiseElement = (event: any): void => {
        const listCopy = expertiseList.slice();
        listCopy.forEach((expertise) => {
            if (expertise.value === event.target.value) {
                expertise.isChecked = event.target.checked;
            }
        });
        setExpertiseList(listCopy);
    };

    return (
        <div className="blog-filter" data-testid="blog-filter">
            <span className="blog-filter__span">Choose a topic youâ€™re interested the most</span>
            <div className="blog-filter__wrapper">
                <div className="blog-filter__list">
                    <span className="blog-filter__list-title">Services</span>
                    {servicesList.map((services) => {
                        return (
                            <FilterList
                                handleCheckChieldElement={handleCheckChieldServicesElement}
                                filterListTitle="Services"
                                {...services}
                            />
                        );
                    })}
                </div>
                <div className="blog-filter__list">
                    <span className="blog-filter__list-title">Industries</span>
                    {industriesList.map((industries) => {
                        return (
                            <FilterList
                                filterListTitle="Industries"
                                {...industries}
                                handleCheckChieldElement={handleCheckChieldIndustriesElement}
                            />
                        );
                    })}
                </div>
                <div className="blog-filter__list">
                    <span className="blog-filter__list-title">Expertise</span>
                    {expertiseList.map((expertise) => {
                        return (
                            <FilterList
                                filterListTitle="Expertise"
                                {...expertise}
                                handleCheckChieldElement={handleCheckChieldExpertiseElement}
                            />
                        );
                    })}
                </div>
            </div>
        </div>
    );
};

export default BlogFilter;
