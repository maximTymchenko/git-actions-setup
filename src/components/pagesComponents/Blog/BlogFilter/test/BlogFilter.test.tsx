import React from 'react';
import BlogFilter from '../BlogFilter';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

it('renders elem correctly', () => {
    const { getByTestId } = render(<BlogFilter />);
    expect(getByTestId('blog-filter')).toBeTruthy();
});

it('renders without unexpected changes', () => {
    const tree = renderer.create(<BlogFilter />).toJSON();
    expect(tree).toMatchSnapshot();
});
