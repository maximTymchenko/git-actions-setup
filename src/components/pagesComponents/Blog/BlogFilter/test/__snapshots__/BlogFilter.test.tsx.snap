// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<div
  className="blog-filter"
  data-testid="blog-filter"
>
  <span
    className="blog-filter__span"
  >
    Choose a topic youâ€™re interested the most
  </span>
  <div
    className="blog-filter__wrapper"
  >
    <div
      className="blog-filter__list"
    >
      <span
        className="blog-filter__list-title"
      >
        Services
      </span>
      <div
        className="filter-list"
        data-testid="filter-list"
      >
        <div
          className="filter-list__body"
        >
          <div
            className="filter-list__chose"
          >
            <label
              className="filter-list__label"
            >
              <input
                checked={false}
                className="filter-list__check"
                onClick={[Function]}
                type="checkbox"
                value="Web development"
              />
              <span
                class="filter-list__checkmark"
              />
              Web development
            </label>
          </div>
        </div>
      </div>
      <div
        className="filter-list"
        data-testid="filter-list"
      >
        <div
          className="filter-list__body"
        >
          <div
            className="filter-list__chose"
          >
            <label
              className="filter-list__label"
            >
              <input
                checked={false}
                className="filter-list__check"
                onClick={[Function]}
                type="checkbox"
                value="Cloud services"
              />
              <span
                class="filter-list__checkmark"
              />
              Cloud services
            </label>
          </div>
        </div>
      </div>
      <div
        className="filter-list"
        data-testid="filter-list"
      >
        <div
          className="filter-list__body"
        >
          <div
            className="filter-list__chose"
          >
            <label
              className="filter-list__label"
            >
              <input
                checked={false}
                className="filter-list__check"
                onClick={[Function]}
                type="checkbox"
                value="Research"
              />
              <span
                class="filter-list__checkmark"
              />
              Research
            </label>
          </div>
        </div>
      </div>
      <div
        className="filter-list"
        data-testid="filter-list"
      >
        <div
          className="filter-list__body"
        >
          <div
            className="filter-list__chose"
          >
            <label
              className="filter-list__label"
            >
              <input
                checked={false}
                className="filter-list__check"
                onClick={[Function]}
                type="checkbox"
                value="Analytics"
              />
              <span
                class="filter-list__checkmark"
              />
              Analytics
            </label>
          </div>
        </div>
      </div>
      <div
        className="filter-list"
        data-testid="filter-list"
      >
        <div
          className="filter-list__body"
        >
          <div
            className="filter-list__chose"
          >
            <label
              className="filter-list__label"
            >
              <input
                checked={false}
                className="filter-list__check"
                onClick={[Function]}
                type="checkbox"
                value="Quality control"
              />
              <span
                class="filter-list__checkmark"
              />
              Quality control
            </label>
          </div>
        </div>
      </div>
      <div
        className="filter-list"
        data-testid="filter-list"
      >
        <div
          className="filter-list__body"
        >
          <div
            className="filter-list__chose"
          >
            <label
              className="filter-list__label"
            >
              <input
                checked={false}
                className="filter-list__check"
                onClick={[Function]}
                type="checkbox"
                value="Mobile development"
              />
              <span
                class="filter-list__checkmark"
              />
              Mobile development
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      className="blog-filter__list"
    >
      <span
        className="blog-filter__list-title"
      >
        Industries
      </span>
      <div
        className="filter-list"
        data-testid="filter-list"
      >
        <div
          className="filter-list__body"
        >
          <div
            className="filter-list__chose"
          >
            <label
              className="filter-list__label"
            >
              <input
                checked={false}
                className="filter-list__check"
                onClick={[Function]}
                type="checkbox"
                value="Retail & E-commerce"
              />
              <span
                class="filter-list__checkmark"
              />
              Retail & E-commerce
            </label>
          </div>
        </div>
      </div>
      <div
        className="filter-list"
        data-testid="filter-list"
      >
        <div
          className="filter-list__body"
        >
          <div
            className="filter-list__chose"
          >
            <label
              className="filter-list__label"
            >
              <input
                checked={false}
                className="filter-list__check"
                onClick={[Function]}
                type="checkbox"
                value="Construction"
              />
              <span
                class="filter-list__checkmark"
              />
              Construction
            </label>
          </div>
        </div>
      </div>
      <div
        className="filter-list"
        data-testid="filter-list"
      >
        <div
          className="filter-list__body"
        >
          <div
            className="filter-list__chose"
          >
            <label
              className="filter-list__label"
            >
              <input
                checked={false}
                className="filter-list__check"
                onClick={[Function]}
                type="checkbox"
                value="Finance"
              />
              <span
                class="filter-list__checkmark"
              />
              Finance
            </label>
          </div>
        </div>
      </div>
      <div
        className="filter-list"
        data-testid="filter-list"
      >
        <div
          className="filter-list__body"
        >
          <div
            className="filter-list__chose"
          >
            <label
              className="filter-list__label"
            >
              <input
                checked={false}
                className="filter-list__check"
                onClick={[Function]}
                type="checkbox"
                value="Media"
              />
              <span
                class="filter-list__checkmark"
              />
              Media
            </label>
          </div>
        </div>
      </div>
      <div
        className="filter-list"
        data-testid="filter-list"
      >
        <div
          className="filter-list__body"
        >
          <div
            className="filter-list__chose"
          >
            <label
              className="filter-list__label"
            >
              <input
                checked={false}
                className="filter-list__check"
                onClick={[Function]}
                type="checkbox"
                value="Other"
              />
              <span
                class="filter-list__checkmark"
              />
              Other
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      className="blog-filter__list"
    >
      <span
        className="blog-filter__list-title"
      >
        Expertise
      </span>
      <div
        className="filter-list"
        data-testid="filter-list"
      >
        <div
          className="filter-list__body"
        >
          <div
            className="filter-list__chose"
          >
            <label
              className="filter-list__label"
            >
              <input
                checked={false}
                className="filter-list__check"
                onClick={[Function]}
                type="checkbox"
                value="SaaS product"
              />
              <span
                class="filter-list__checkmark"
              />
              SaaS product
            </label>
          </div>
        </div>
      </div>
      <div
        className="filter-list"
        data-testid="filter-list"
      >
        <div
          className="filter-list__body"
        >
          <div
            className="filter-list__chose"
          >
            <label
              className="filter-list__label"
            >
              <input
                checked={false}
                className="filter-list__check"
                onClick={[Function]}
                type="checkbox"
                value="Chatbots"
              />
              <span
                class="filter-list__checkmark"
              />
              Chatbots
            </label>
          </div>
        </div>
      </div>
      <div
        className="filter-list"
        data-testid="filter-list"
      >
        <div
          className="filter-list__body"
        >
          <div
            className="filter-list__chose"
          >
            <label
              className="filter-list__label"
            >
              <input
                checked={false}
                className="filter-list__check"
                onClick={[Function]}
                type="checkbox"
                value="AR & VR"
              />
              <span
                class="filter-list__checkmark"
              />
              AR & VR
            </label>
          </div>
        </div>
      </div>
      <div
        className="filter-list"
        data-testid="filter-list"
      >
        <div
          className="filter-list__body"
        >
          <div
            className="filter-list__chose"
          >
            <label
              className="filter-list__label"
            >
              <input
                checked={false}
                className="filter-list__check"
                onClick={[Function]}
                type="checkbox"
                value="Blockchain"
              />
              <span
                class="filter-list__checkmark"
              />
              Blockchain
            </label>
          </div>
        </div>
      </div>
      <div
        className="filter-list"
        data-testid="filter-list"
      >
        <div
          className="filter-list__body"
        >
          <div
            className="filter-list__chose"
          >
            <label
              className="filter-list__label"
            >
              <input
                checked={false}
                className="filter-list__check"
                onClick={[Function]}
                type="checkbox"
                value="Digital enterprise "
              />
              <span
                class="filter-list__checkmark"
              />
              Digital enterprise 
            </label>
          </div>
        </div>
      </div>
      <div
        className="filter-list"
        data-testid="filter-list"
      >
        <div
          className="filter-list__body"
        >
          <div
            className="filter-list__chose"
          >
            <label
              className="filter-list__label"
            >
              <input
                checked={false}
                className="filter-list__check"
                onClick={[Function]}
                type="checkbox"
                value="Drone"
              />
              <span
                class="filter-list__checkmark"
              />
              Drone
            </label>
          </div>
        </div>
      </div>
      <div
        className="filter-list"
        data-testid="filter-list"
      >
        <div
          className="filter-list__body"
        >
          <div
            className="filter-list__chose"
          >
            <label
              className="filter-list__label"
            >
              <input
                checked={false}
                className="filter-list__check"
                onClick={[Function]}
                type="checkbox"
                value="R&D"
              />
              <span
                class="filter-list__checkmark"
              />
              R&D
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
